docs/pages/guides/server-components.mdx

```diff
@@ -0,0 +1,395 @@
+
+```tsx components/button.tsx
+'use client';
+import { Text } from 'react-native';
+
+export default function Button({ title }) {
+  return <Text onPress={() => {}}>{title}</Text>;
+}
+```
+
+This file can be imported and used in a Server Component:
+
+```tsx app/index.tsx
+import Button from '../components/button';
+
+export default function Index() {
+  return <Button title="Press me" />;
+}
+```
+
+You cannot pass functions as props to Server Components. You can only pass serializable data.
+
+## Server Actions
+
+> Server Actions are gated behind an additional experimental flag. Pass the environment variable `EXPO_UNSTABLE_SERVER_ACTIONS=1` to enable them.
+
+Server Actions are functions that run on the server and can be called from client components. Think of them like fully-typed API routes that are easier to write.
+
+Server Actions must always be an async function and are marked with `"use server"` at the top of the function.
+
+```tsx app/index.tsx
+export default function Index() {
+  return (
+    <Button
+      title="Press me"
+      onPress={async () => {
+        'use server';
+        // This code runs on the server.
+        console.log('Button pressed');
+        return '...';
+      }}
+    />
+  );
+}
+```
+
+You can create a client component to invoke the server action:
+
+```tsx components/button.tsx
+'use client';
+import { Text } from 'react-native';
+
+export default function Button({ title, onPress }) {
+  return <Text onPress={() => onPress()}>{title}</Text>;
+}
+```
+
+Server Actions can also be defined in a standalone file (with `"use server"` at the top) and imported from client components:
+
+```tsx components/server-actions.tsx
+'use server';
+
+export async function callAction() {
+  // ...
+}
+```
+
+These can be used in client components:
+
+```tsx components/button.tsx
+import { Text } from 'react-native';
+import { callAction } from './server-actions';
+
+export default function Button({ title }) {
+  return <Text onPress={() => callAction()}>{title}</Text>;
+}
+```
+
+- You can only pass serializable data to server actions as arguments.
+- Server actions can only return serializable data.
+- Server actions run on the server and therefore are a good place to put logic that should not be exposed to the client.
+
+## CSS
+
+Expo Router supports importing global CSS and CSS modules in server components.
+
+```tsx app/index.tsx
+import './styles.css';
+import styles from './styles.module.css';
+
+export default function Index() {
+  return <div className={styles.container}>Hello, world!</div>;
+}
+```
+
+The CSS will be hoisted into the client bundle from the server.
+
+## Metadata
+
+React Server Components are a feature of React 19. To enable them, Expo CLI automatically uses a special canary build of React on all platforms. We'll remove this feature in the future when React 19 is enabled by default in React Native.
```

Comment:
```suggestion
React Server Components are a feature of React 19. To enable them, Expo CLI automatically uses a special canary build of React on all platforms. In the future, it will be removed when React 19 is enabled by default in React Native.
```